File: 2020/csharp/Aoc2020.Tests/Day12Tests.cs
using System.Linq;
using System.Collections.Generic;
using Xunit;

namespace Aoc2020.Tests
{
    public class Day12Tests
    {
        [Fact]
        public void MoveWorks()
        {
            Assert.Equal(
                25,
                Day12.ManhattanDistance(
                    Utils.ReadInputLines("12-test")
                    .Select(Movement.Parse)
                    .Aggregate((0, 0, 0), Day12.Move)));
        }

        [Fact]
        public void MoveWaypointWorks()
        {
            Assert.Equal(
                286,
                Day12.ManhattanDistance(
                    Utils.ReadInputLines("12-test")
                    .Select(Movement.Parse)
                    .Aggregate(((0, 0), (10, 1)), Day12.MoveWaypoint)));
        }

        [Fact]
        public void Stars()
        {
            var run = new Day12();
            Assert.Equal("445", run.Result1());
            Assert.Equal("42495", run.Result2());
        }
    }
}

File: 2016/fsharp/Aoc2016.Tests/Day12.Tests.fs
module Day12.Tests

open Xunit
open Utils
open Day12

[<Fact>]
let ``Day12 computer works`` () =
    let input = [| "cpy 41 a"; "inc a"; "inc a"; "dec a"; "jnz a 2"; "dec a" |]

    Assert.Equal(
        42<value>,
        Computer.empty |> Computer.load input |> Computer.run |> Computer.get RA
    )

[<Fact>]
let ``Day12 Stars`` () =
    try
        Assert.Equal("318020", job1 ())
        Assert.Equal("9227674", job2 ())
    with :? System.NotImplementedException ->
        ()
